driver-core/src/main/com/mongodb/connection/ByteBufferBsonOutput.java

```diff
@@ -177,7 +177,9 @@ public void truncateToPosition(final int newPosition) {
     @Override
     public void close() {
         for (final ByteBuf cur : bufferList) {
-            cur.release();
+            while(cur.getReferenceCount() > 0) {
```

Comment:
This is pretty unusual, as normally any code that retains a reference is responsible for releasing that reference.  Any idea who's not releasing properly?